React/Views/RCTView.m

```diff
@@ -716,6 +716,21 @@ - (NSString *)description
 #endif // TODO(macOS GH#774)
 
 #if TARGET_OS_OSX // [TODO(macOS GH#774)
+- (NSShadow*)shadow {
+  CALayer *layer = self.layer;
+
+  // Workaround AppKit issue with directly manipulating the view layer's shadow.
+  if (layer.shadowColor != nil && layer.shadowOpacity > 0) {
+    NSShadow *shadow = [NSShadow new];
```

Comment:
Slight style choice to only have one return, which we prefer: https://microsoft.github.io/objc-guide/MethodsAndImplementations/EarlyReturns.html :

```objective-c

NSShadow *shadow = nil
if (layer.shadowColor != nil && layer.shadowOpacity > 0) {
  shadow = [NSShadow new];
  ...
}
return shadow
```